src/tito/builder/submodule_aware_builder.py

```diff
@@ -0,0 +1,210 @@
+# Copyright (c) 2008-2011 Red Hat, Inc.
+#
+# This software is licensed to you under the GNU General Public License,
+# version 2 (GPLv2). There is NO WARRANTY for this software, express or
+# implied, including the implied warranties of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+# along with this software; if not, see
+# http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+#
+# Red Hat trademarks are not licensed under GPLv2. No permission is
+# granted to use or replicate Red Hat trademarks that are incorporated
+# in this software or its documentation.
+"""
+Tito builders for a variety of common methods of building sources, srpms,
+and rpms.
+"""
+
+import os
+
+from tito.builder import Builder
+from tito.common import debug, run_command, create_tgz, find_spec_like_file, get_commit_timestamp
+from tito.tar import TarFixer
+
+
+class PushDir(object):
```

Comment:
I think we have this already implemented as `tito.common.chdir`. For example `MeadBuilder` uses it as

    with chdir(self.maven_clone_dir):
        run_command("git checkout %s" % self.git_commit_id)

Reply:
Oh interesting, I hadn't seen this.

Lemme try and figure that part out and issue it instead.

Reply:
This is perfect, I'll use with chdir